private-media-framework/draft-ietf-perc-private-media-framework.md

```diff
@@ -711,3 +698,362 @@ a co-author.
+|           | The salt to use with this key is provided by the   |
+|           | key distributor via the EKT Key message.  All      |
+|           | endpoints use the same E2E master salt.            |
++-----------+----------------------------------------------------+
+~~~
+Figure: Key Inventory
+
+As you can see, the number key types is very small.  However, what
+can be challenging is keeping track of all of the distinct E2E keys as
+the conference grows in size.  With 1,000 participants in a conference,
+there will be 1,000 distinct SRTP master keys, all of which share the
+same master salt.  Each of those keys are passed through the KDF defined
+in [@RFC3711] to produce the actual encryption and authentication keys.
+Complicating key management is the fact that the KEK can change and,
+when it does, the endpoints generate new SRTP master keys.  And, of
+course, there is a new SRTP master salt to go with those keys.
+Endpoints have to retain old keys for a period of time to ensure they
+can properly decrypt late-arriving or out-of-order packets.
+
+Or more detailed explanation of each of the keys follows.
+
+## DTLS-SRTP Exchange Yields HBH Keys
+
+The first set of keys acquired are for hop-by-hop encryption and
+decryption.  Assuming the use of Double [@!I-D.ietf-perc-double], the
+endpoint would perform DTLS-SRTP exchange with the key distributor
+and receive a key that is, in fact, "double" the size that is needed.
+Per the Double specification, the E2E part is the first half of the key,
+so the endpoint will just discard that information in PERC.  It is not
+used.  The second half of the key material is for HBH operations, so
+that half of the key is assigned internally as the HBH key.
+
+The media distributor doesn't perform DTLS-SRTP, but it is at this point
+that the key distributor will inform the media distributor of the HBH key
+value via the tunnel protocol ([@!I-D.ietf-perc-dtls-tunnel]).
+No two endpoints will have the same HBH key, thus the media distributor
+must keep track each distinct HBH key and use it only for the
+corresponding hop.
+
+Note that the SRTP master salt used for HBH encryption is computed
+along with they key.  Likewise, the salt value is transmitted to
+the media distributor.
+
+This key is also used for HBH encryption of RTCP.  RTCP is not
+end-to-end encrypted in PERC.
+
+## The Key Distributor Transmits the KEK (EKT Key)
+
+Via the aforementioned DTLS-SRTP association, the key distributor will
+send the endpoint the KEK (i.e., EKT Key per
+[@!I-D.ietf-perc-srtp-ekt-diet]).  This key is known only to
+the key distributor and endpoints.  This key is the most important to
+protect since having knowledge of this key will allow an entity to
+decrypt any media packet in the conference.
+
+Note that the key distributor can send any number of EKT Keys to
+endpoints.  This can be used to re-key the entire conference.  Each
+key is identified by a "Security Parameter Index" (SPI) value.
```

Comment:
add something along lines of it is common to re-key the whole conference when a participant leaves the conference